src/cameras.jl

```diff
@@ -51,150 +103,34 @@ end
 
 Camera centered over all vehicles, does not change the zoom level.
 """
-struct SceneFollowCamera <: Camera end
-function update_camera!(rendermodel::RenderModel, camera::SceneFollowCamera, scene::Frame{E}) where {E<:Entity}
+struct SceneFollowCamera <: Camera
+    state::CameraState
+end
+SceneFollowCamera(;kwargs...) = SceneFollowCamera(CameraState(;kwargs...))
+function update_camera!(camera::SceneFollowCamera, scene::Frame{E}) where {E<:Entity}
     C = sum([posg(veh.state)[1:2] for veh in scene])/length(scene)  # center of mass
-    set_camera!(rendermodel, x=C[1], y=C[2])
+    set_camera!(camera.cs, x=C[1], y=C[2])
+    # should also add capabilities for adapting zoom level to make all vehicles of the scene fit
+    # along the lines of FitToContentCamera, but much simpler (just using entity coordinates to determine bounding box but INCLUDING zoom)
 end
 
 
 """
 Composition of several cameras. The `update_camera` actions of the individual cameras are applied in the order in which they are saved in the `cameras` array.
+States of individual cameras are ignored, the state of the composed camera is the one that will be used for rendering.
 
 Example Usage
 
     cam = ComposedCamera(cameras=[SceneFollowCamera(), ZoomingCamera()])
 """
-@with_kw mutable struct ComposedCamera <: Camera
+mutable struct ComposedCamera <: Camera
+    state::CameraState
     cameras::Array{Camera}
 end
+ComposedCamera(cameras; kwargs...) = ComposedCamera(CameraState(;kwargs...), cameras)
 
-function update_camera!(rendermodel::RenderModel, camera::ComposedCamera, scene::Frame{E}) where {E<:Entity}
+function update_camera!(camera::ComposedCamera, scene::Frame{E}) where {E<:Entity}
     for cam in camera.cameras
-        update_camera!(rendermodel, cam, scene)
-    end
-end
-
-
-"""
-Positions the camera so that all content is visible within its field of view
-This will always set the camera rotation to zero
-An extra border can be added as well
-"""
-@with_kw mutable struct FitToContentCamera <: Camera
-    canvas_width::Int64 = DEFAULT_CANVAS_WIDTH
-    canvas_height::Int64 = DEFAULT_CANVAS_HEIGHT
-    percent_border::Float64 = 0.1
-end
-function update_camera!(rendermodel::RenderModel, cam::FitToContentCamera, scene::Frame{E}) where {E<:Entity}
-    camera_fit_to_content!(rendermodel, cam.canvas_width, cam.canvas_height, percent_border=cam.percent_border)
-    rendermodel
-end
```

Comment:
`FitToContentCamera` can no longer be supported because it relies on accessing the instructions of the `rendermodel` to compute the zoom level. `camera_fit_to_content` function is still there and can be used as a replacement.

It would be nice to adapt the `SceneFollowCamera` in a way that it also allows to optionally adapt the zoom level to fit all vehicles of the scene, see notes above.

Reply:
Just made another commit that includes adaptive zoom for SceneFollowCamera. I assume this would make `FitToContentCamera` more or less redundant, but I'm not exactly aware of the use cases...

Reply:
sounds good, the code for `FitToContent` seems weirdly specific.

Reply:
@MaximeBouton would you say "weirdly specific" enough to get rid of it in v0.8? Or should we leave it around?

Reply:
It is still a useful feature let's keep it?